core/src/test/scala/unit/kafka/log/LogSegmentsTest.scala

```diff
@@ -105,8 +109,6 @@ class LogSegmentsTest {
     assertFalse(segments.nonEmpty)
     assertEquals(0, segments.numberOfSegments)
     assertFalse(segments.contains(offset1))
-
-    segments.close()
```

Comment:
why removing this line?

Reply:
Invoking `segments.close()` is redundant here since `segments.clear()` already empties the values in segments.

Reply:
You're right. So it is a bug here. Now, I think what we have to do is manually close seg1, seg2, seg3 here, otherwise, there will be resource leak.